broker-core/src/main/java/io/zeebe/broker/workflow/processor/CatchEventBehavior.java

```diff
@@ -237,11 +246,24 @@ private boolean sendOpenMessageSubscription(
         workflowInstanceKey, elementInstanceKey, messageName, correlationKey, closeOnCorrelate);
   }
 
-  public class MessageCorrelationKeyException extends RuntimeException {
-    private static final long serialVersionUID = 8929284049646192937L;
-
-    MessageCorrelationKeyException(String message) {
-      super(message);
+  private Map<DirectBuffer, DirectBuffer> extractMessageCorrelationKeys(
+      List<ExecutableCatchEvent> events,
+      MessageCorrelationKeyContext elementContext,
+      MessageCorrelationKeyContext scopeContext) {
+    extractedCorrelationKeys.clear();
+
+    for (ExecutableCatchEvent event : events) {
+      if (event.isMessage()) {
+        final MessageCorrelationKeyContext context =
+            event.getElementType() == BpmnElementType.BOUNDARY_EVENT
+                ? scopeContext
+                : elementContext;
+        final DirectBuffer correlationKey = extractCorrelationKey(event.getMessage(), context);
+
+        extractedCorrelationKeys.put(event.getId(), cloneBuffer(correlationKey));
+      }
     }
+
+    return extractedCorrelationKeys;
```

Comment:
Why do you return this, if it's a field?

Reply:
It's a field at the moment to avoid re-allocating it every time, but the design is more flexible if when I use it I don't care about that, just that some method returns to the me the extracted correlation keys.